src/components/EditableInput/EditableInput.tsx

```diff
@@ -4,78 +4,156 @@ import React, {
     FocusEventHandler,
     KeyboardEvent,
     KeyboardEventHandler,
-    ReactNode,
+    useEffect,
     useRef,
     useState,
 } from 'react';
+import { EditableInputHelper } from '@components/EditableInput/lib/helper';
+import { DEFAULT_CONTAINER_CLASS, DEFAULT_INPUT_TEXT_CLASS } from '@components/EditableInput/constant';
 
+export enum EditableMode {
+    INPUT = 'INPUT',
+    LABEL = 'LABEL',
+}
+
+/**
+ * Define custom Styles with:
+ * customContainerClasses (<div>)
+ * customInputTextClasses (<input>)
+ * mode: display as Input or Label first
+ * enableDoubleClick: Should the input be triggered by a double click or single click
+ * additionalValues: additional information to be passed via onClick Event
+ */
+interface EditableOptionProps {
+    customContainerClasses?: string;
+    customInputTextClasses?: string;
+    mode?: EditableMode;
+    enableDoubleClick?: boolean;
+    additionalValues?: string;
+}
+
+/**
+ * Add Children with badges or icons
+ * onEditableSave callback to get updated Value
+ * onAdditionalValueSave callback with more information (ex. ID)
+ * onModeChange callback when state changes
+ * options to specify styling and additional behaviours
+ */
 export interface EditableInputProps {
-    name: string;
-    targetItemId: string;
-    onEditableSave: (targetItemId: string, value: string) => void;
-    children?: ReactNode;
+    children?: React.ReactElement;
+    onAdditionalValueSave?: (additionalValue: string, value: string) => void;
+    onEditableSave?: (value: string) => void;
+    onModeChange?: (editableState?: EditableMode) => void;
+    options?: EditableOptionProps;
 }
 
-export const EditableInput = ({ name, targetItemId, onEditableSave, children }: EditableInputProps) => {
-    const [inputValue, setInputValue] = useState(name);
-    const [showInput, setShowInput] = useState<boolean>(false);
+/**
+ * Component to switch between Label and Input state
+ *
+ * Clones the children and adds clickHandlers to flip on Click.
+ * Events:
+ * - onEditableSave         // When changing from input to label state
+ * - onModeChange           // Indicates which state is active
+ * - onAdditionalValueSave  // Similar to editable Save with additional Values provided
+ *
+ * Options properties:
+ * - customContainerClasses: string             // Tailwind Classes
+ * - customInputTextClasses: string             // Tailwind Classes
+ * - mode:                   enum EditableMode  // {INPUT = 'INPUT',LABEL = 'LABEL'}
+ * - enableDoubleClick:      boolean            // should the input be triggered by a double click or single click
+ * - additionalValues:       string             // additional information to be passed via onClick Event
+ *
+ * @param onEditableSave
+ * @param onModeChange
```

Comment:
I'm personally not a big fan of JsDocs in React components, it looks noisy and redundant to name the params when we have interfaces. The code speaks of itself. And we don't really use it anywhere else. 🤔 But we can ask what others prefer.